### ISSUE TITLE
Remove per-actor event loop

### ISSUE THREAD

**Issue by m1sterc001guy** [fedimint/fedimint#2192]
Today, our lightning gateway supports two node implementations: CLN and LND. The main responsibility of CLN and LND for receiving payments from the gateway's perspective is to provide intercepted HTLCs to `gatewayd` for processing.

After https://github.com/fedimint/fedimint/pull/2161, for both CLN and LND, there will be one thread that is responsible for "intercepting" HTLCs. For CLN, this thread exists inside the cln_extension and for LND it is a separate thread inside `gatewayd`. 

For both implementations, this thread then looks up a `Sender` object in a map that it uses to send the HTLC to the core business logic of `gatewayd`. The key of this map is a short_channel_id, which corresponds to a single actor inside `gatewayd` For CLN extension, the interceptor thread waits for a response from the actor's event loop which processes this HTLC. For LND, the interceptor thread does not wait for a response, but there is a `stream` setup which is passed to LND and that listens for a `ForwardInterceptedHtlcResponse` from `gatewayd`.

The obersvation here is that, we currently have N actor event loops that are waiting for intercepted HTLCs from CLN or LND. However, the interception of the HTLCs is already single threaded, so it is very likely that most of these actor event loops are simply waiting around with nothing to do. This seems to be a waste of resources.

The proposal is to refactor each actor's event loop into a single event loop that is managed by `gatewayd`. Each `actor` can be refactored to be a single data structure that represents a federation client. This new single event loop can simply wait for intercepted HTLCs from the interceptor thread, lookup the actor in a data structure, process the HTLC, then pass it back to the lightning implementation.

The main advantage here is just simplifying the code. We would now only have 2 event loops: one for intercepting HTLCs and one for processing HTLCs. I don't think we loose much in terms of performance, since we are already bottlenecked by the interceptor thread.

Thoughts?

**Comment by dpc** [comment]
Note: I barely even know what's going on there, so just going to say some generally applicable platitudes. :D

Simple is better, but there's a question of blocking and latency. We need multiple threads/tasks if we want to have multiple handling procedures progress in paralle. Otherwise with two requests arriving roughly at the same time, one might need to wait for the other to finish.

I doubt one task channel is the right thing to do anyway. We could either have some fixed amount of threads/tasks (our max parallelism), or spawn a new task on each request

But! Then there's a question of locking and sharing global resources. If every handler processing in parallel somehow only needs only one-channel related things, there might be a reason to have them per-channel. But I doubt that's the case.

**Comment by m1sterc001guy** [comment]
Ah I think I missed that the CLN extension is actually spawning a new thread per intercepted HTLC. This makes it worth having a separate thread per actor. I'll close this.

**Comment by justinmoon** [comment]
My feeling is that the "actor" is an abstraction that we don't really benefit from. If it could just be " gateway federation client" without any event loop, that be a win in terms of making the code easier to reason about.

Regarding performance concerns, I think we need some benchmarks to inform those decisions.
